bin/bedrock/src/args.rs

```diff
@@ -0,0 +1,134 @@
+use std::path::PathBuf;
+
+use bedrock_server::{
+    Config,
+    ConfigError,
+    ConfigFile,
+    StandardConfigFile,
+};
+use clap::{
+    ArgAction,
+    Parser,
+};
+use si_service::prelude::*;
+
+const NAME: &str = "bedrock";
+
+pub(crate) fn parse() -> Args {
+    Args::parse()
+}
+
+#[derive(Parser, Debug)]
+#[command(name = NAME, max_term_width = 100)]
+pub(crate) struct Args {
+    /// Sets the verbosity mode.
+    ///
+    /// Multiple -v options increase verbosity. The maximum is 6.
+    #[arg(short = 'v', long = "verbose", action = ArgAction::Count)]
+    pub(crate) verbose: u8,
+
+    /// Disables ANSI coloring in log output, even if standard output refers to a terminal/TTY.
+    ///
+    /// For more details, visit: <http://no-color.org/>.
+    #[arg(
+        long = "no-color",
+        default_value = "false",
+        env = "SI_NO_COLOR",
+        hide_env_values = true,
+        conflicts_with = "force_color"
+    )]
+    pub(crate) no_color: bool,
+
+    /// Forces ANSI coloring, even if standard output refers to a terminal/TTY.
+    ///
+    /// For more details, visit: <http://no-color.org/>.
+    #[arg(
+        long = "force-color",
+        default_value = "false",
+        env = "SI_FORCE_COLOR",
+        hide_env_values = true,
+        conflicts_with = "no_color"
+    )]
+    pub(crate) force_color: bool,
+
+    /// Prints telemetry logging as JSON lines.
+    ///
+    /// For more details, visit: <https://jsonlines.org/>.
+    #[arg(
+        long = "log-json",
+        default_value = "false",
+        env = "SI_LOG_JSON",
+        hide_env_values = true
+    )]
+    pub(crate) log_json: bool,
+
+    /// The address and port to bind the HTTP server to [example: 0.0.0.0:80]
+    #[arg(long, env)]
+    pub(crate) socket_addr: Option<String>,
+
+    // Section for NATS configuration
+    /// NATS connection URL [example: demo.nats.io]
+    #[arg(long)]
+    pub(crate) nats_url: Option<String>,
+
+    /// NATS credentials string
+    #[arg(long, allow_hyphen_values = true)]
+    pub(crate) nats_creds: Option<SensitiveString>,
+
+    /// NATS credentials file
+    #[arg(long)]
+    pub(crate) nats_creds_path: Option<PathBuf>,
+
+    /// AWS Credentials
+    #[arg(long, env = "AWS_SECRET_ACCESS_KEY", hide_env_values(true))]
+    aws_secret_access_key: Option<String>,
+    #[arg(long, env = "AWS_ACCESS_KEY_ID", hide_env_values(true))]
+    aws_access_key_id: Option<String>,
+    #[arg(long, env = "AWS_SESSION_TOKEN", hide_env_values(true))]
+    aws_session_token: Option<String>,
```

Comment:
I know this is already merged, but this is really smelly to me. The aws_config bits should be able to pull this info out of the environment. Are we planning to pass credentials in as flags?